### PR TITLE
[ios, macos] Bumped Xcode version requirement to 8.0

### PR DESCRIPTION
**Pull request by jmkiley**
v3.5.2 does not seem to be compatible with Xcode 7.3.1 and crashes. Bumping the Xcode version requirement to 8.0.

cc @mapbox/ios

### CODE REVIEW

File: platform/ios/CHANGELOG.md [hunk 1]
```diff
@@ -18,6 +18,7 @@ Mapbox welcomes participation and contributions from everyone. Please read [CONT
 * The error passed into `-[MGLMapViewDelegate mapViewDidFailLoadingMap:withError:]` now includes a more specific description and failure reason. ([#8418](https://github.com/mapbox/mapbox-gl-native/pull/8418))
 * Fixed an issue where gesture recognizers associated with map view interactivity were not disabled when their related interactions were disabled. ([#8304](https://github.com/mapbox/mapbox-gl-native/pull/8304))
 * Fixed an issue where re-adding a layer that had been previously removed from a style would reset its paint properties. Moved initializers for `MGLTileSource`, `MGLStyleLayer`, and `MGLForegroundStyleLayer` to their concrete subclasses; because these classes were already intended for initialization only via concrete subclasses, this should have no developer impact. ([#8626](https://github.com/mapbox/mapbox-gl-native/pull/8626))
+* Xcode 8.0 or higher is now required to use this SDK.
```
Comment (1ec5) [inline platform/ios/CHANGELOG.md]: This is important enough that we should make it the first item for the release.

**Comment by friedbunny** [comment]
The crash while using Xcode 7.3.1 with v3.5.2 occurs while loading dylibs:

```
* thread #1: tid = 0x2c5239, 0x000000010a9162fe dyld_sim`memcmp + 9, stop reason = EXC_BAD_ACCESS (code=50, address=0x10a981000)
  * frame #0: 0x000000010a9162fe dyld_sim`memcmp + 9
    frame #1: 0x000000010a90a33b dyld_sim`ImageLoaderMachO::validateFirstPages(linkedit_data_command const*, int, unsigned char const*, unsigned long, long long, ImageLoader::LinkContext const&) + 121
    frame #2: 0x000000010a90cee2 dyld_sim`ImageLoaderMachOCompressed::instantiateFromFile(char const*, int, unsigned char const*, unsigned long, unsigned long long, unsigned long long, stat const&, unsigned int, unsigned int, linkedit_data_command const*, encryption_info_command const*, ImageLoader::LinkContext const&) + 290
    frame #3: 0x000000010a909643 dyld_sim`ImageLoaderMachO::instantiateFromFile(char const*, int, unsigned char const*, unsigned long long, unsigned long long, stat const&, ImageLoader::LinkContext const&) + 327
    frame #4: 0x000000010a8fde8d dyld_sim`dyld::loadPhase6(int, stat const&, char const*, dyld::LoadContext const&) + 456
    frame #5: 0x000000010a9014f1 dyld_sim`dyld::loadPhase5(char const*, char const*, dyld::LoadContext const&, std::__1::vector<char const*, std::__1::allocator<char const*> >*) + 599
    frame #6: 0x000000010a901280 dyld_sim`dyld::loadPhase4(char const*, char const*, dyld::LoadContext const&, std::__1::vector<char const*, std::__1::allocator<char const*> >*) + 180
    frame #7: 0x000000010a900c95 dyld_sim`dyld::loadPhase3(char const*, char const*, dyld::LoadContext const&, std::__1::vector<char const*, std::__1::allocator<char const*> >*) + 762
    frame #8: 0x000000010a90074c dyld_sim`dyld::loadPhase1(char const*, char const*, dyld::LoadContext const&, std::__1::vector<char const*, std::__1::allocator<char const*> >*) + 135
    frame #9: 0x000000010a8fdc9a dyld_sim`dyld::loadPhase0(char const*, char const*, dyld::LoadContext const&, std::__1::vector<char const*, std::__1::allocator<char const*> >*) + 213
    frame #10: 0x000000010a8fda28 dyld_sim`dyld::load(char const*, dyld::LoadContext const&) + 179
    frame #11: 0x000000010a901623 dyld_sim`dyld::libraryLocator(char const*, bool, char const*, ImageLoader::RPathChain const*) + 52
    frame #12: 0x000000010a906ee0 dyld_sim`ImageLoader::recursiveLoadLibraries(ImageLoader::LinkContext const&, bool, ImageLoader::RPathChain const&, char const*) + 270
    frame #13: 0x000000010a906458 dyld_sim`ImageLoader::link(ImageLoader::LinkContext const&, bool, bool, bool, ImageLoader::RPathChain const&) + 92
    frame #14: 0x000000010a8fef01 dyld_sim`dyld::link(ImageLoader*, bool, bool, ImageLoader::RPathChain const&) + 149
    frame #15: 0x000000010a8fff05 dyld_sim`dyld::_main(macho_header const*, unsigned long, int, char const**, char const**, char const**, unsigned long*) + 2809
    frame #16: 0x000000010a8fc251 dyld_sim`start_sim + 136
    frame #17: 0x0000000110499510 dyld`dyld::useSimulatorDyld(int, macho_header const*, char const*, int, char const**, char const**, char const**, unsigned long*, unsigned long*) + 2246
    frame #18: 0x0000000110497a39 dyld`dyld::_main(macho_header const*, unsigned long, int, char const**, char const**, char const**, unsigned long*) + 241
    frame #19: 0x0000000110493249 dyld`dyldbootstrap::start(macho_header const*, int, char const**, long, macho_header const*, unsigned long*) + 470
    frame #20: 0x0000000110493036 dyld`_dyld_start + 54
```

**Comment by 1ec5** [comment]
Previously https://github.com/mapbox/mapbox-gl-native/issues/6842#issuecomment-257726195.

**Comment by friedbunny** [comment]
Confirmed that Xcode 8.0 still works.

**Comment by jmkiley** [comment]
Confirmed that the **-symbols** build worked with Xcode 7.3.1. I will revise to document that you should use that version of the framework with Xcode 7.3.1.

File: platform/macos/CHANGELOG.md [hunk 2]
```diff
@@ -10,6 +10,7 @@
 
 ### Other changes
 
+* Xcode 8.0 or higher is now recommended for using this SDK. ([#8775(https://github.com/mapbox/mapbox-gl-native/pull/8775)======
```
Comment (friedbunny) [inline platform/macos/CHANGELOG.md]: Seems like a few extra `=` snuck in here.

**Pull request merged**